ui/jstests/test_manage_taxonomies.jsx

```diff
@@ -91,4 +345,348 @@ define(['QUnit', 'jquery', 'setup_manage_taxonomies', 'reactaddons',
         );
     }
   );
+
+  QUnit.test('Assert that AddTermsComponent work properly',
+    function(assert) {
+      assert.ok(AddTermsComponent, "class object not found");
+      var vocabularies = [
+        {
+          "vocabulary": vocabulary,
+          "terms": vocabulary.terms
+        }
+      ];
+      var done = assert.async();
+      var addTermCalled = 0;
+      var afterMount = function(component) {
+        assertAddTermCommon(assert, component);
+        var inputGroup = React.addons.TestUtils.
+          findRenderedDOMComponentWithClass(
+            component,
+            'input-group'
+          );
+        var textbox = React.addons.TestUtils.
+          findRenderedDOMComponentWithTag(
+            inputGroup,
+            'input'
+          );
+        React.addons.TestUtils.Simulate.keyUp(textbox, {key: "Enter"});
+        waitForAjax(1, function() {
+          //test items
+          assert.equal(addTermCalled, 1);
+          done();
+        });
+      };
+      var addTerm = function() {
+        addTermCalled += 1;
+      };
+      var addTermsComponentRendered = React.addons.TestUtils.
+        renderIntoDocument(
+          <AddTermsComponent
+            vocabularies={vocabularies}
+            repoSlug="repo"
+            addTerm={addTerm}
+            ref={afterMount}
+          />
+        );
+      assert.equal(
```

Comment:
Please move this code into the `afterMount` function so we ensure it executes after the component is mounted